src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -1333,7 +1333,7 @@ void CompositionalMultiphaseBase::applySourceFluxBC( real64 const time,
   {
     fsManager.apply< ElementSubRegionBase >( time + dt,
                                              mesh,
-                                             FieldSpecificationBase::viewKeyStruct::fluxBoundaryConditionString(),
+                                             SourceFluxBoundaryCondition::catalogName(),
```

Comment:
so instead of this, we would template the `FieldSpecification` type to `fsManager.apply<SourceFluxBoundaryCondition>()`. Would that be cleaner?

Reply:
Done, now it is `fsManager.apply< ElementSubRegionBase, SourceFluxBoundaryCondition >`